### PR TITLE
Ability to disable scheduler and controller-manager

### LINKED ISSUE #84: [kwokctl] Ability to disable `scheduler` and `controller-manager`
_Originally posted by @wzshiming in https://github.com/kubernetes-sigs/kube-scheduler-simulator/issues/251#issuecomment-1327140339_

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

the cluster created by kwokctl should not start the scheduler and kube-scheduler-simulator should take care of the scheduling, so that kwok should be able to be integrated.

#### Which issue(s) this PR fixes:

Fixes #84 

#### Does this PR introduce a user-facing change?

```release-note
kwokctl: Ability to disable scheduler and controller-manager
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by carlory** [comment]
cc @wzshiming

File: pkg/kwokctl/vars/vars.go [hunk 1]
```diff
@@ -76,6 +76,12 @@ var (
 	// QuietPull is the flag to quiet the pull.
 	QuietPull = getEnvBool("KWOK_QUIET_PULL", false)
 
+	// DisableKubeScheduler is the flag to disable kube-scheduler.
+	DisableKubeScheduler = getEnvBool("DISABLE_KUBE_SCHEDULER", false)
```
Comment (wzshiming) [inline pkg/kwokctl/vars/vars.go:80]:
```suggestion
	DisableKubeScheduler = getEnvBool("KWOK_DISABLE_KUBE_SCHEDULER", false)
```

File: pkg/kwokctl/vars/vars.go [hunk 2]
```diff
@@ -76,6 +76,12 @@ var (
...
+
+	// DisableKubeControllerManager is the flag to disable kube-controller-manager.
+	DisableKubeControllerManager = getEnvBool("DISABLE_KUBE_CONTROLLER_MANAGER", false)
```
Comment (wzshiming) [inline pkg/kwokctl/vars/vars.go:83]:
```suggestion
	DisableKubeControllerManager = getEnvBool("KWOK_DISABLE_KUBE_CONTROLLER_MANAGER", false)
```

File: pkg/kwokctl/runtime/kind/cluster.go [hunk 3]
```diff
@@ -223,6 +223,27 @@ func (c *Cluster) Up(ctx context.Context) error {
...
 	c.Kubectl(ctx, utils.IOStreams{}, "config", "set", "contexts."+conf.Name+".user", "kind-"+conf.Name)
+
+	// waiting for kwok-controller and other addons to be ready
+	// Fixme: use a better way to wait for pods to be ready when the kube-controller-manager is disabled
+	err = c.Kubectl(ctx, utils.IOStreams{
+		ErrOut: os.Stderr,
+	}, "-n", "kube-system", "wait", "deployment/kwok-controller", "--for", "condition=Available=true", "--timeout=5m")
+	if err != nil {
+		return err
+	}
```
Comment (wzshiming) [inline pkg/kwokctl/runtime/kind/cluster.go:234]:
```suggestion
	if conf.DisableKubeControllerManager {
		// waiting for kwok-controller and other addons to be ready
		// Fixme: use a better way to wait for pods to be ready when the kube-controller-manager is disabled
		err = c.Kubectl(ctx, utils.IOStreams{
			ErrOut: os.Stderr,
		}, "-n", "kube-system", "wait", "deployment/kwok-controller", "--for", "condition=Available=true", "--timeout=5m")
		if err != nil {
			return err
		}
	}
```

**Comment by carlory** [comment]
cc @wzshiming

**Pull request merged**